Remove AppConstants

kean opened the pull request:
Self-explanatory.

To test:

- Both apps don't crash
- Verify line-by-line that the values were moved accurately, please

## Regression Notes
1. Potential unintended areas of impact


2. What I did to test those areas of impact (or what existing automated tests I relied on)


3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

kean committed:
Remove AppConstants.AboutScreen

kean committed:
Remove AppConstants.AppRatings

kean committed:
Remove shareTitle

kean committed:
Replace Settings

kean committed:
Remove AppConstants.Settings

kean committed:
Remove AppConstants.Logout

kean committed:
Replace productTwitterHandle

kean committed:
Replace productTwitterURL

kean committed:
Replace productBlogURL

kean committed:
Replace productBlogDisplayURL

kean committed:
Replace zendeskSourcePlatform

kean committed:
Replace ShareAppName

kean committed:
Remove mobileAnnounceAppId

kean committed:
Replace authKeychainServiceName

kean committed:
Remove AppConstants

crazytonyli commented on Modules/Sources/BuildSettingsKit/BuildSettings+Live.swift:
```diff
@@ -13,6 +13,14 @@ extension BuildSettings {
+        )
+        zendeskSourcePlatform = bundle.infoValue(forKey: "WPZendeskSourcePlatform")
+        mobileAnnounceAppID = bundle.infoValue(forKey: "WPMobileAnnounceAppID")
+        authKeychainServiceName = bundle.infoValue(forKey: "WPAuthKeychainServiceName")
```
All of these are just hard-coded values which don't have anything to do with build configurations. Maybe we can just have some readonly variables or something? I don't think all these values need to be in the Info.plist files.

mokagio replied:
> All of these are just hard-coded values which don't have anything to do with build configurations

True. However, they do change from WordPress and Jetpack. Maybe we could derive them from `brand`/`WPAppBrand`. But then we'd have sort of two places where to place the values: `Info.plist` and the code `switch brand { case ...` here.

kean replied:
>  two places where to place the values: Info.plist and the code switch brand { case ... here.

💯. The idea is to specify configuration in data: `Info.plist` and `.xcconfig`. It's not always feasible, but you generally shouldn't have to search the codebase to find these values – it should be in configuration files, in data. I used `switch brand` sparingly.

There are multiple advantages to storing these values in data and not code: easier to find, can be modified without recompiling the app, easier to edit with scripts if needed.

mokagio commented:
I think I know why the builds are failing...

---

Update: Finally got CI validation https://github.com/wordpress-mobile/WordPress-iOS/pull/24297

mokagio committed:
Remove `https://` from `Info.plist`s

Apparently, Xcode/`xcodebuild` fails to parse it, see how
https://buildkite.com/automattic/wordpress-ios/builds/26578/steps?jid=0195bb59-1f1a-40ad-89c9-3964c36cff28#0195bb59-1f1a-40ad-89c9-3964c36cff28/973-5502
failed and notice the build for this commit passes.

kean committed:
Remove `https://` from `Info.plist` to address compilation fail (#24297)

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-iOS/milestone/308">25.9</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

kean committed:
Merge branch 'trunk' into task/remove-app-constants

kean committed:
Merge branch 'trunk' into task/remove-app-constants

pmusolino commented on WordPress/Info.plist:
```diff
@@ -20,8 +20,20 @@
+	<key>WPProductBlogURL</key>
+	<string>wordpress.org/news/</string>
+	<key>WPZendeskSourcePlatform</key>
+	<string>mobile_-_ios</string>
```
Should this be `mobile_-_wp_ios`? Is there a reason why we do not identify the app?

kean replied:
I copied the value from https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/WordPress/Classes/Utility/App%20Configuration/AppConstants.swift:

```swift
 static let zendeskSourcePlatform = "mobile_-_ios"
```

pmusolino reviewed (approved):
I just did a general smoke test, and I didn't find any crashes 👍
